lib/api/multiObjectDelete.js

```diff
@@ -271,9 +266,25 @@ export function getObjMetadataAndDelete(authInfo, canonicalID, request,
             // TODO: update number of objects to reflect creation of
             // delete markers; Utapi must be updated to allow this for
             // 'deleteObject' metrics
-            const newDeleteMarker = deleteInfo.newDeleteMarker;
-            successfullyDeleted.push({ entry, newDeleteMarker,
-                isDeletingDeleteMarker: objMD.isDeleteMarker });
+            let isDeleteMarker;
+            let deleteMarkerVersionId;
+            // If trying to delete an object that does not exist (empty objMD)
+            // return DeleteMarkerVersionId equals the versionID of the
+            // marker you just generated and DeleteMarker tag equals true
+            if (deleteInfo.newDeleteMarker) {
+                isDeleteMarker = true;
+                deleteMarkerVersionId = versionIdUtils.encode(versionId);
+            // If trying to delete a delete marker, DeleteMarkerVersionId equals
+            // deleteMarker's versionID and DeleteMarker equals true
+            } else if (objMD && objMD.isDeleteMarker) {
+                isDeleteMarker = true;
+                deleteMarkerVersionId = entry.versionId;
+            } else {
+                isDeleteMarker = false;
+                deleteMarkerVersionId = undefined;
```

Comment:
default value of `deleteMarkerVersionId` should already be undefined and if you don't care if `isDeleteMarker` is `undefined` instead of `false`, you don't need this condition block. It's up to you though, maybe you want to be explicit.